### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by orianacolmenares**
- Create SecurityConfiguration, CustomAuthenticationProvider and PasswordEncoderService files to validate access to Users and books information. 
- Add Password attribute to Users model. 
- Add spring-security dependencies.

**Trello Card:** https://trello.com/c/pty6fmqA/23-integrar-spring-security

### CODE REVIEW

File: src/main/java/wolox/training/config/CustomAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        String password = authentication.getCredentials().toString();
+
+        Optional<Users> user = usersRepository.findByUsername(name);
+
+        boolean matchPassword = PasswordEncoderService.passwordMatch(password, user.get().getPassword());
+
+        if (user.isPresent() && matchPassword) {
+            return new UsernamePasswordAuthenticationToken(
+                    name, password, new ArrayList<>());
+        } else {
```
Comment (IgnOliveto) [inline src/main/java/wolox/training/config/CustomAuthenticationProvider.java:35]: Why dont throw a custom exception here so we can tell what was wrong here??

Comment (orianacolmenares) [reply src/main/java/wolox/training/config/CustomAuthenticationProvider.java:35]: I don't know why or how, but it already do it. Before when I have the matches issue, the response was very clear about what was failing. But if its better practice I can do the custom exception.

Comment (IgnOliveto) [reply src/main/java/wolox/training/config/CustomAuthenticationProvider.java:35]: Is ok, the default exception here can work, you can always create an exception or exception handler.

File: src/main/java/wolox/training/models/Users.java [hunk 2]
```diff
@@ -35,9 +39,10 @@ public Users(long id) {
         this.books = new ArrayList<>();
     }
 
-    public Users(String username, String name, LocalDate birthday) {
+    public Users(String username, String name, String password, LocalDate birthday) {
         this.username = username;
         this.name = name;
+        this.password = password;
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/models/Users.java]: Here you should use the setter. If you use this constructor as it is now, then you can create an User without an encoded password

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]
Well done! There you go some comments :) 
Besides that, I just noticed that your test classes are `BookRestControllerIntegrationTest` and `UserRestControllerIntegrationTest`, the word `Integration` there has no sense. Because what you are doing is a Unit test, you are not integrating all the application and testing it (The training card `Test - Controllers y Daos` explains the difference)
So, the correct names for those classes would be just `BookControllerTest` and `UserControllerTest`
Even the word `Rest` could be avoided, because it the controller could change on the future and use another protocol

**Review by leandrocostawolox — APPROVED** [review]

**Pull request merged**